### DISCUSSION TITLE
Support for KIP-86 SASL callback handlers

### DISCUSSION THREAD

**Discussion by robertcoltheart** [edenhill/librdkafka#4359] [Ideas]
Hi @edenhill,

I'd like to contribute a PR to cover KIP-86, but wanted first to check if this idea had already been discussed? If so, was there a specific reason why it hasn't been implemented to date?

Let me know if you're open to a PR to cover this functionality.

**Comment by milindl** [comment]
There is  `rd_kafka_sasl_set_credentials`, which serves the same purpose. (#4033)
There's no plan to implement it in a callback form, since we have this in place.

**Reply by robertcoltheart** [reply]
Forgive my ignorance, but this only supports username and password, correct? It won't support more complex scenarios like MSK IAM authentication.
